jit/v2/JIT_v2.0.0.yaml

```
@@ -174,59 +174,55 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorResponse'
+  '/v2/port-call-services':
     get:
       parameters:
-        - $ref: '#/components/parameters/portCallServiceIDPathParam'
+        - $ref: '#/components/parameters/portCallIDQueryParam'
+        - $ref: '#/components/parameters/transportCallIDQueryParam'
+        - $ref: '#/components/parameters/portCallServiceIDQueryParam'
+
+        - $ref: '#/components/parameters/startDateTimeQueryParam'
+        - $ref: '#/components/parameters/endDateTimeQueryParam'
+
+        - $ref: '#/components/parameters/UNLocationCodeQueryParameter'
+        - $ref: '#/components/parameters/portVisitReferenceQueryParam'
+        - $ref: '#/components/parameters/carrierServiceNameQueryParam'
+        - $ref: '#/components/parameters/carrierServiceCodeQueryParam'
+        - $ref: '#/components/parameters/universalServiceReferenceQueryParam'
+
+        - $ref: '#/components/parameters/transportCallReferenceQueryParam'
+        - $ref: '#/components/parameters/transportCallSequenceNumberQueryParam'
+        - $ref: '#/components/parameters/carrierImportVoyageNumberQueryParam'
+        - $ref: '#/components/parameters/carrierExportVoyageNumberQueryParam'
+        - $ref: '#/components/parameters/universalImportVoyageReferenceQueryParam'
+        - $ref: '#/components/parameters/universalExportVoyageReferenceQueryParam'
+        - $ref: '#/components/parameters/portCallServiceTypeQueryParam'
+        - $ref: '#/components/parameters/portCallServiceEventTypeCodeQueryParam'
+
+        - $ref: '#/components/parameters/vesselIMONumberQueryParam'
+        - $ref: '#/components/parameters/vesselNameQueryParam'
+        - $ref: '#/components/parameters/MMSINumberQueryParam'
+
+        - $ref: '#/components/parameters/limitQueryParam'
+        - $ref: '#/components/parameters/cursorQueryParam'
         - $ref: '#/components/parameters/Api-Version-Major'
       tags:
         - Port Call Service
-      operationId: get-port-call-service
-      summary: Retrieves a Port Call Service
+      operationId: get-port-call-services
+      summary: Retrieves a list of Port Call Services
       description: |
-        Retrieves a Port Call Service
+        Retrieves a list of Port Call Services
```

```suggestion
        Retrieve a list of Port Call Services.
```
Additional question to the above changes: 

1.  Are no filter parameters mandatory? That would probably be an overload of info. 
2. If many parameters are used to filter with, the URL gets very long, given the long names. Perhaps it can get too long. If many need to be applied together, we might should consider to move to POST and accept it as a body request.

---

1. Since there is a default limit of 100 `Port Call Services` - personally I don't think we should limit what consumers want. In other of our APIs - business wants to force-limit the response payload and therefore each request **must** be limited by a set of filters... @emiliodcsa does business have any concerns about the resulting payload?
2. Also a suggestion from @gj0dcsa in the Teams chat. We have not previously done this. I think the "normal" limit of modern browsers is about 2K characters...

---

Description modified :-)